src/extension.ts

```diff
@@ -49,8 +49,9 @@ async function getClusterKubeconfig(target: AksClusterTreeItem): Promise<string
     }
     const client = new azcs.ContainerServiceClient(target.root.credentials, target.root.subscriptionId);  // TODO: safely
     try {
-        const accessProfile = await client.managedClusters.getAccessProfile(resourceGroupName, name, 'clusterUser');
-        const kubeconfig = accessProfile.kubeConfig!.toString();  // TODO: safely
+        const clusterUserCredentials = await client.managedClusters.listClusterUserCredentials(resourceGroupName, name);
+        const kubeconfigList = clusterUserCredentials.kubeconfigs!.filter((kubeInfo) => kubeInfo.name === "clusterUser");  // TODO: safely
```

Comment:
I don't _think_ you need the `!` here (which is what the 'TODO: safely' refers to).  The docs imply you always get the array - though obviously if the TS bindings type it as nullable then we have to work to those.

Reply:
Consider using `kubeconfigs.find(...)` instead of `kubeconfigs.filter(...)[0]` (and then check for `undefined`).